File: src/main/java/xyz/oribuin/skyblock/managers/MessageManager.java
package xyz.oribuin.skyblock.managers;

import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import xyz.oribuin.skyblock.Skyblock;
import xyz.oribuin.skyblock.utils.FileUtils;
import xyz.oribuin.skyblock.utils.StringPlaceholders;

import java.io.File;

public class MessageManager extends Manager {

    private final static String MESSAGE_CONFIG = "messages.yml";

    private FileConfiguration messageConfig;

    public MessageManager(Skyblock plugin) {
        super(plugin);
    }

    @Override
    public void reload() {
        FileUtils.createFile(this.plugin, MESSAGE_CONFIG);
        this.messageConfig = YamlConfiguration.loadConfiguration(new File(this.plugin.getDataFolder(), MESSAGE_CONFIG));
    }

    public void sendMessage(CommandSender sender, String messageId) {
        this.sendMessage(sender, messageId, StringPlaceholders.empty());
    }

    public void sendMessage(CommandSender sender, String messageId, StringPlaceholders placeholders) {
        sender.sendMessage(ChatColor.translateAlternateColorCodes('&', placeholders.apply(this.messageConfig.getString(messageId))));
    }
}

File: src/main/kotlin/xyz/oribuin/skyblock/manager/MessageManager.kt
package xyz.oribuin.skyblock.manager

import net.md_5.bungee.api.ChatMessageType
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import xyz.oribuin.skyblock.Skyblock
import xyz.oribuin.skyblock.hook.PlaceholderAPIHook
import xyz.oribuin.skyblock.library.HexUtils.colorify
import xyz.oribuin.skyblock.library.Manager
import xyz.oribuin.skyblock.library.StringPlaceholders
import xyz.oribuin.skyblock.library.StringPlaceholders.Companion.empty

class MessageManager(plugin: Skyblock) : Manager(plugin) {

    override fun reload() {
        // Unused
    }


    fun sendMessage(sender: CommandSender, message: String, placeholders: StringPlaceholders = empty()) {
        sender.spigot().sendMessage(*TextComponent.fromLegacyText(placeholders.apply(colorify(parsePlaceholders(sender, message)))))
    }

    fun sendActionMessage(sender: Player, message: String, placeholders: StringPlaceholders = empty()) {
        sender.spigot().sendMessage(ChatMessageType.ACTION_BAR, *TextComponent.fromLegacyText(placeholders.apply(colorify(parsePlaceholders(sender, message)))))
    }

    private fun parsePlaceholders(sender: CommandSender, message: String): String {
        return if (sender is Player)
            PlaceholderAPIHook.apply(sender, message)
        else
            message
    }

    override fun disable() {
        // Unused
    }
}
